### PR TITLE
Feature/line y axis scaling

### PR DESCRIPTION
**Pull request by olslash**
Adds "hasMultipleYAxes" option for line widgets and visualizations.  When set to true, the min and max of the second graph line is scaled to match the first, and a second y-axis is rendered with the scale of the second line.

@drakonen @kaareal

### CODE REVIEW

File: bower.json [hunk 1]
```diff
@@ -31,7 +31,7 @@
     "moment": "~2.8.3",
     "moment-timezone": "~0.2.2",
     "requirejs-text": "~2.0.12",
-    "rickshaw": "https://github.com/drakonen/rickshaw.git#v1.5.3"
```
Comment (drakonen) [inline bower.json]: Seems like a minor version update, not a patch.

File: dist/openhose-visualizations-sdk.bundle.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
 /*
```
Comment (drakonen) [inline dist/openhose-visualizations-sdk.bundle.js]: Only do bundling in master. Makes Pull Requests easier.

File: docs/examples/visualizations/line.html [hunk 3]
```diff
@@ -34,6 +34,64 @@
...
+                "end":      1404943199999,
+                "timezone": 'Europe/Amsterdam',
+                "bucket":   "1d"
+              });
+
+              var data = {
+                values: [{
+                  "color":          "#1F77B4",
+                  "data":           [
+                    {"x": 1404165600, "y": 31454},
```
Comment (drakonen) [inline docs/examples/visualizations/line.html]: can you make the numbers smaller? say 100 - 125 ?

File: src/visualization/lib/dots.js [hunk 4]
```diff
@@ -39,7 +49,7 @@ define(['core', 'rickshaw'], function (Openhose, Rickshaw) {
             return self.graph.x(d.x);
           })
           .attr("cy", function (d) {
-            return self.graph.y(d.y + d.y0);
+            return yCalc(d.y || 0);
```
Comment (drakonen) [inline src/visualization/lib/dots.js]: Why did you remove the dy0 ?

File: src/visualization/line.js [hunk 5]
```diff
@@ -116,7 +150,7 @@ define([
         var annotationData = annotations.data;
         var style = annotations.style;
         var labelStyle = annotations.labelStyle;
-        this.$el.append('<div class="annotations"/></div>');
```
Comment (drakonen) [inline src/visualization/line.js]: nicely spotted

File: src/visualization/line.js [hunk 6]
```diff
@@ -167,21 +201,42 @@ define([
...
+          orientation: 'right',
+          scale:       data[0].scale
+        });
+        var axis;
+        //I declare this here before the axis as the axis itself will call
+        //graph.onUpdate and we need to set height and width on axis before that happens
+        //other wise it wont render the axis correctly, translate(undefined, NaN)
+        this.graph.onUpdate(function () { 
+          axis.height = self.graph.height;
+          axis.width = self.graph.width;
```
Comment (drakonen) [inline src/visualization/line.js]: does that need a re-render?

Comment (kaareal) [inline src/visualization/line.js]: it does not

File: dist/openhose-visualizations-sdk.bundle.js (hunk 2 above)
Comment (olslash) [inline dist/openhose-visualizations-sdk.bundle.js]: :heavy_check_mark:  makes sense.

File: docs/examples/visualizations/line.html (hunk 3 above)
Comment (olslash) [inline docs/examples/visualizations/line.html]: done

File: src/visualization/lib/dots.js (hunk 4 above)
Comment (olslash) [inline src/visualization/lib/dots.js]: looks like kaare fixed this.

File: bower.json (hunk 1 above)
Comment (olslash) [inline bower.json]: gonna switch to openhose/rickshaw, once we merge the couple PRs on there we can resolve by bumping it a minor version

**Comment by drakonen** [comment]
+1

**Comment by drakonen** [comment]
When creating a new release, it will also contain "Added temporalchange viewOption. Disabled by default." https://github.com/openhose/openhose-visualizations-sdk/pull/50

**Comment by olslash** [comment]
I got to the bottom of the race issue-- either something's wrong with chrome or with grunt dev.

I noticed that even with a clean branch and running grunt-dev, i would see multiple axes working about half the time and half not. I put a debugger into the `if(hasMultipleYAxes)` in  visualization/line and noticed that it was only hitting my breakpoint on times when the graph was rendering correctly -- it would be skipped otherwise.

I had changed the flag from 'hasMultipleYAxis' to 'hasMultipleYAxes' a few commits ago, but for some reason half the time the file was being loaded with the first and half with the second. So the examples were setting 'axes' and the file was loading with 'axis'.

In any case, I think we're good to go since i checked the bundle and it's correct.
